File: DMOJ/ccc96s1.py
def kek(n):
  return sum([i if n % i == 0 else 0 for i in range(1, n)])

for i in range(int(input())):
  n = int(input())
  gg = kek(n)
  if gg == n:
    print(str(n) + " is a perfect number.")
  elif gg > n:
    print(str(n) + " is an abundant number.")
  else:
    print(str(n) + " is a deficient number.")

File: DMOJ/ccc96s1.c
#import <stdio.h>
#import <math.h>

int x;

int main(void) {
    scanf("%i", &x);
    for (int j = 0; j < x; j++) {
        int n;
        int sum = 1;
        scanf("%i", &n);
        for (int i = 2; i <= sqrt(n); i++) {
            if (i == sqrt(n)) {
                sum += i;
            } else if (n % i == 0) sum += i + n/i;
        }
        if (sum > n) {
            printf("%i is an abundant number.\n", n);
        } else if (sum < n) {
            printf("%i is a deficient number.\n", n);
        } else {
            printf("%i is a perfect number.\n", n);
        } 
    }

}